### ISSUE TITLE
Cannot establish tunnel to Ubuntu 22.04 server

### ISSUE THREAD

**Issue by fm89** [dbeaver/dbeaver#16852]
#### System information: Version	

- Microsoft Windows 10 Pro (10.0.18363 Build 18363)
- DBeaver 22.1.0.202206051253 (June 6, 2022)
- No additional extension

#### Connection specification:
- Database : PostgreSQL 14.3
- Driver : org.postgresql:postgresql:RELEASE 42.2.5
- SSH Tunnel used

#### Describe the problem you're observing:

I (try to) use an SSH Tunnel but I cannot make it work. When I click on "Test tunnel configuration", I get the message "**Cannot establish tunnel to [XX.XX.XX.XX]:22**" and "**Auth fail**".

![image](https://user-images.githubusercontent.com/47399735/174794388-1553311f-d225-4a1b-9d61-b0120d097e8f.png)

I am familiar with this procedure and I use it very often for Ubuntu 18.04 and 20.04 servers. Here, my new server runs Ubuntu 22.04 and I cannot make it work.

I have checked using other tools that I can indeed establish the SSH Tunnel with my new server. For example, the database tool of PhpStorm connects successfully to this database, with the same configuration for the tunnel.

I include the full logs below, could you please tell me if this is a bug or what to try next?

#### Include any warning/errors/backtraces from the logs
```
> Start DBeaver SSH tunnels UI [org.jkiss.dbeaver.net.ssh.ui 1.0.104.202206051253]
> Start DBeaver SSH tunnels [org.jkiss.dbeaver.net.ssh 1.0.104.202206051253]
2022-06-21 13:52:35.922 - Service 'org.jkiss.dbeaver.runtime.ui.UIServiceSecurity' not found
> Start Java Native Access [com.sun.jna 5.8.0.v20210503-0343]
> Start DBeaver SSH implementation (JSch) [org.jkiss.dbeaver.net.ssh.jsch 1.1.93.202206051253]
2022-06-21 13:52:37.818 - Adding identity key
2022-06-21 13:52:37.834 - Configure tunnel
> Start Noyau JSch [org.eclipse.jsch.core 1.3.900.v20200422-1935]
> Start Interface utilisateur JSch [org.eclipse.jsch.ui 1.4.100.v20210917-1201]
2022-06-21 13:52:37.849 - Connect to tunnel host
2022-06-21 13:52:37.849 - SSH INFO: Connecting to [EDITED-FOR-PRIVACY] port 22
2022-06-21 13:52:37.919 - SSH INFO: Connection established
2022-06-21 13:52:37.950 - SSH INFO: Remote version string: SSH-2.0-OpenSSH_8.9p1 Ubuntu-3
2022-06-21 13:52:37.950 - SSH INFO: Local version string: SSH-2.0-JSCH-0.1.54
2022-06-21 13:52:37.950 - SSH INFO: CheckCiphers: aes256-ctr,aes192-ctr,aes128-ctr,aes256-cbc,aes192-cbc,aes128-cbc,3des-ctr,arcfour,arcfour128,arcfour256
2022-06-21 13:52:37.950 - SSH INFO: CheckKexes: diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521
2022-06-21 13:52:38.003 - SSH INFO: CheckSignatures: ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
2022-06-21 13:52:38.003 - SSH INFO: SSH_MSG_KEXINIT sent
2022-06-21 13:52:38.019 - SSH INFO: SSH_MSG_KEXINIT received
2022-06-21 13:52:38.019 - SSH INFO: kex: server: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,sntrup761x25519-sha512@openssh.com,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256
2022-06-21 13:52:38.019 - SSH INFO: kex: server: rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519
2022-06-21 13:52:38.019 - SSH INFO: kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
2022-06-21 13:52:38.019 - SSH INFO: kex: server: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com
2022-06-21 13:52:38.019 - SSH INFO: kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
2022-06-21 13:52:38.019 - SSH INFO: kex: server: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1
2022-06-21 13:52:38.019 - SSH INFO: kex: server: none,zlib@openssh.com
2022-06-21 13:52:38.019 - SSH INFO: kex: server: none,zlib@openssh.com
2022-06-21 13:52:38.019 - SSH INFO: kex: server: 
2022-06-21 13:52:38.019 - SSH INFO: kex: server: 
2022-06-21 13:52:38.019 - SSH INFO: kex: client: ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group14-sha1,diffie-hellman-group-exchange-sha256,diffie-hellman-group-exchange-sha1,diffie-hellman-group1-sha1
2022-06-21 13:52:38.019 - SSH INFO: kex: client: ssh-rsa,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521
2022-06-21 13:52:38.019 - SSH INFO: kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc,aes192-ctr,aes192-cbc,aes256-ctr,aes256-cbc
2022-06-21 13:52:38.019 - SSH INFO: kex: client: aes128-ctr,aes128-cbc,3des-ctr,3des-cbc,blowfish-cbc,aes192-ctr,aes192-cbc,aes256-ctr,aes256-cbc
2022-06-21 13:52:38.019 - SSH INFO: kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
2022-06-21 13:52:38.019 - SSH INFO: kex: client: hmac-md5,hmac-sha1,hmac-sha2-256,hmac-sha1-96,hmac-md5-96
2022-06-21 13:52:38.019 - SSH INFO: kex: client: none
2022-06-21 13:52:38.019 - SSH INFO: kex: client: none
2022-06-21 13:52:38.019 - SSH INFO: kex: client: 
2022-06-21 13:52:38.019 - SSH INFO: kex: client: 
2022-06-21 13:52:38.019 - SSH INFO: kex: server->client aes128-ctr hmac-sha1 none
2022-06-21 13:52:38.019 - SSH INFO: kex: client->server aes128-ctr hmac-sha1 none
2022-06-21 13:52:38.019 - SSH INFO: SSH_MSG_KEX_ECDH_INIT sent
2022-06-21 13:52:38.019 - SSH INFO: expecting SSH_MSG_KEX_ECDH_REPLY
2022-06-21 13:52:38.066 - SSH INFO: Host '[EDITED-FOR-PRIVACY]' is known and matches the ECDSA host key
2022-06-21 13:52:38.066 - SSH INFO: SSH_MSG_NEWKEYS sent
2022-06-21 13:52:38.066 - SSH INFO: SSH_MSG_NEWKEYS received
2022-06-21 13:52:38.066 - SSH INFO: SSH_MSG_SERVICE_REQUEST sent
2022-06-21 13:52:38.104 - SSH INFO: SSH_MSG_SERVICE_ACCEPT received
2022-06-21 13:52:38.166 - SSH INFO: Authentications that can continue: publickey,keyboard-interactive,password
2022-06-21 13:52:38.172 - SSH INFO: Next authentication method: publickey
2022-06-21 13:52:38.220 - SSH INFO: Disconnecting from [EDITED-FOR-PRIVACY] port 22
2022-06-21 13:52:38.320 - Auth fail
com.jcraft.jsch.JSchException: Auth fail
	at com.jcraft.jsch.Session.connect(Session.java:519)
	at com.jcraft.jsch.Session.connect(Session.java:183)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationJsch.setupTunnel(SSHImplementationJsch.java:116)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.initTunnel(SSHImplementationAbstract.java:134)
	at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.initializeHandler(SSHTunnelImpl.java:76)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$7.execute(SSHTunnelConfiguratorUI.java:247)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$TunnelConnectionTestJob.run(SSHTunnelConfiguratorUI.java:593)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by E1izabeth** [comment]
Hi, @fm89! Could you, please, try to switch to SSHJ in Advanced settings? Does it help?
![image](https://user-images.githubusercontent.com/28875055/175003295-4a36154b-d716-4e4d-aa60-40b52ec29b48.png)

**Comment by fm89** [comment]
Thanks for considering this report.

When I select SSHJ and click "Test Connection", I get another error : *Cannot establish tunnel / Timeout expired*.

The log file contains the following errors

```
2022-06-22 12:31:49.644 - No such algorithm: IDEA/CBC/NoPadding
2022-06-22 12:31:49.644 - No such algorithm: IDEA/CTR/NoPadding
2022-06-22 12:31:49.648 - Disabling high-strength ciphers: cipher strengths apparently limited by JCE policy
2022-06-22 12:31:49.722 - Client identity string: SSH-2.0-SSHJ_0.33.0
2022-06-22 12:31:49.760 - Server identity string: SSH-2.0-OpenSSH_8.9p1 Ubuntu-3
Exception in thread "sshj-Reader-/[EDITED-FOR-PRIVACY]:22" java.lang.NoClassDefFoundError: com/hierynomus/asn1/encodingrules/ASN1Encoder
	at net.schmizz.sshj.signature.SignatureECDSA$Factory256.create(SignatureECDSA.java:40)
	at net.schmizz.sshj.signature.SignatureECDSA$Factory256.create(SignatureECDSA.java:36)
	at com.hierynomus.sshj.key.BaseKeyAlgorithm.newSignature(BaseKeyAlgorithm.java:58)
	at net.schmizz.sshj.transport.kex.AbstractDHG.next(AbstractDHG.java:82)
	at net.schmizz.sshj.transport.KeyExchanger.handle(KeyExchanger.java:388)
	at net.schmizz.sshj.transport.TransportImpl.handle(TransportImpl.java:485)
	at net.schmizz.sshj.transport.Decoder.decode(Decoder.java:113)
	at net.schmizz.sshj.transport.Decoder.received(Decoder.java:200)
	at net.schmizz.sshj.transport.Reader.run(Reader.java:60)
Caused by: java.lang.ClassNotFoundException: com.hierynomus.asn1.encodingrules.ASN1Encoder cannot be found by org.jkiss.bundle.sshj_0.33.0
	at org.eclipse.osgi.internal.loader.BundleLoader.generateException(BundleLoader.java:529)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:524)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:416)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:168)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 9 more
2022-06-22 12:32:19.818 - Timeout expired
java.util.concurrent.TimeoutException: Timeout expired
	at net.schmizz.concurrent.Promise.retrieve(Promise.java:139)
	at net.schmizz.concurrent.Event.await(Event.java:105)
	at net.schmizz.sshj.transport.KeyExchanger.waitForDone(KeyExchanger.java:148)
	at net.schmizz.sshj.transport.KeyExchanger.startKex(KeyExchanger.java:143)
	at net.schmizz.sshj.transport.TransportImpl.doKex(TransportImpl.java:254)
	at net.schmizz.sshj.SSHClient.doKex(SSHClient.java:825)
	at net.schmizz.sshj.SSHClient.onConnect(SSHClient.java:813)
	at net.schmizz.sshj.SocketClient.connect(SocketClient.java:69)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationSshj.setupTunnel(SSHImplementationSshj.java:84)
	at org.jkiss.dbeaver.model.net.ssh.SSHImplementationAbstract.initTunnel(SSHImplementationAbstract.java:134)
	at org.jkiss.dbeaver.model.net.ssh.SSHTunnelImpl.initializeHandler(SSHTunnelImpl.java:76)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$7.execute(SSHTunnelConfiguratorUI.java:247)
	at org.jkiss.dbeaver.ui.net.ssh.SSHTunnelConfiguratorUI$TunnelConnectionTestJob.run(SSHTunnelConfiguratorUI.java:593)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by E1izabeth** [comment]
Thanks for a quick reply

**Comment by fm89** [comment]
As a side note, to check that everything is OK on my computer and the distant server, I checked that I can connect to the database using DBeaver and a port forwarding by running the following command 

```
ssh -L localhost:5555:localhost:5432 [SERVER-IP-ADDRESS] -i Z:/[PATH-TO-PRIVATE-KEY] -l [MY-LOGIN]
```

Once I run this in a Windows command prompt, I can connect correctly to the distant database by making it point towards localhost:5555.

**Comment by jeremyakers** [comment]
Seeing the exact same issue here. We can tunnel DBeaver via an Ubuntu 20.04 bastion host but not an Ubuntu 22.04 bastion host.

I'm running DBeaver on Linux, so I don't have any other options besides JSch.

Manually creating the tunnel via SSH command works just fine. Just can't do it via Dbeaver.

**Comment by dbeaver-pm** [comment]
Probably we need to add asn1 dependency (https://github.com/hierynomus/asn-one)

**Comment by kret13** [comment]
I have the same issue. Have you found a workaround for this?

**Comment by fm89** [comment]
Well, I do not know a workaround from within DBeaver itself. But you can setup a tunnel manually using the ssh command as mentioned above. Then make DBeaver point towards a local address (which is port-forwarded). Obviously, this is not optimal, as it requires you to perform this additionnal step each time before using DBeaver.

**Comment by rodabt** [comment]
> Seeing the exact same issue here. We can tunnel DBeaver via an Ubuntu 20.04 bastion host but not an Ubuntu 22.04 bastion host.
> 
> I'm running DBeaver on Linux, so I don't have any other options besides JSch.
> 
> Manually creating the tunnel via SSH command works just fine. Just can't do it via Dbeaver.

Same here

**Comment by Ou7law007** [comment]
> > Seeing the exact same issue here. We can tunnel DBeaver via an Ubuntu 20.04 bastion host but not an Ubuntu 22.04 bastion host.
> > I'm running DBeaver on Linux, so I don't have any other options besides JSch.
> > Manually creating the tunnel via SSH command works just fine. Just can't do it via Dbeaver.
> 
> Same here

Same here version 22.1.3

**Comment by mbfdias** [comment]
I have the same problem with dbeaver Version 22.1.3.202207271354 running on an ubuntu 18.04 connecting to just a installed brand new ubuntu 22.04

adding PubkeyAcceptedKeyTypes=+ssh-rsa to the 22.04 server /etc/ssh/sshd_config and restarting sshd temporarily bypasses the problem and allows the tunnel to work

**Comment by TOMHTML** [comment]
Hi,
I had the same issue, DBeaver was unable to use my SSH tunnel while there was no issue with the terminal. More than that, only one tunnel had this issue, we duplicated the server and the ssh tunnel and there was no problem there.

The problem is the server had multiple public keys, and for some reason Dbeaver wasn't using the "good" one, only the key listed in the "known hosts". The terminal command `ssh` handle that characteristic like a charm.

The solution: 
```
ssh-keyscan -H example.com >> ~/.ssh/known_hosts
```
That made DBeaver connection a success without restarting the app 🚀 

Hope that helps

**Comment by DanVal80** [comment]
> The solution: 
> ```
> ssh-keyscan -H example.com >> ~/.ssh/known_hosts
> ```
> That made DBeaver connection a success without restarting the app 🚀

Thank you @TOMHTML. I have been struggling for weeks with this issue. Your solution worked immediately! (Dbeaver 22.1.4 connecting to Ubuntu 22.04 via SSHJ tunnel).

**Comment by serge-rider** [comment]
We will upgrade SSHJ version and add ASN1 dependency in the next version

**Comment by tmas** [comment]
So I'm unsure why SSHing in from a terminal doesn't also encounter this issue, but when the `ssh-keyscan` trick didn't work for me I was able to fix the issue by generating an ed25519 key to replace my existing RSA key.

It looks like Ubuntu 22.04 deprecated rsa-based SSH keys, so I'd say this is probably outside of the project scope. I'm not sure if there's a way to see what version of OpenSSH the server you're trying to connect to is running, but if any change is made to DBeaver to mitigate this issue I'd recommend adding a warning when authenticating to an OpenSSH server that has deprecated RSA support fails and the user is trying to authenticate with an RSA key.

**Comment by fm89** [comment]
Hi,
It now works correctly using SSHJ (not with JSch though).
I would like to thank all the persons involved in this correction!
Thanks for building and maintaining such a great tool.
